demos/ota/ota_demo_core_http/ota_demo_core_http.c

```diff
@@ -1178,23 +1180,34 @@ static int establishConnection( void )
 
     return returnStatus;
 }
+
 static void disconnect( void )
 {
     /* Disconnect from broker. */
     LogInfo( ( "Disconnecting the MQTT connection with %s.", AWS_IOT_ENDPOINT ) );
 
-    if( pthread_mutex_lock( &mqttMutex ) == 0 )
+    if( mqttSessionEstablished == true )
     {
-        /* OTA job is completed. so delete the network connection. */
-        MQTT_Disconnect( &mqttContext );
+        if( pthread_mutex_lock( &mqttMutex ) == 0 )
+        {
+            /* Disconnect MQTT session. */
+            MQTT_Disconnect( &mqttContext );
 
-        pthread_mutex_unlock( &mqttMutex );
+            pthread_mutex_unlock( &mqttMutex );
+
+            /* Clear the mqtt session flag. */
+            mqttSessionEstablished = false;
```

Comment:
May make more sense to have this inside of the mutex after the disconnect call.

Reply:
Addressed in 
https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1451/commits/463ab1e0eb8ae36da14dfb76e6ab62d316ff0ab1